main.py

```diff
@@ -0,0 +1,87 @@
+import operator
+from typing import Callable
+Self = 'Matrix'
+
+
+# Purpose: Create a class to represent a two-dimensional array
+class Matrix:
+    """A class to represent a two-dimensional array"""
+    def __init__(self, matrix: tuple[tuple[float, ...], ...]):
+        self._matrix = matrix
+
+    def __str__(self):
+        """Return a string representation: ((1, 2), (3, 4))"""
+        return str(self._matrix)
+
+    def __repr__(self):
+        """ return: a string representation: Matrix(((1, 2), (3, 4)))"""
+        return f'Matrix({self._matrix})'
+
+    @property
+    def tuples(self):
+        """Return the matrix"""
+        return self._matrix
+
+    def _addition_subtraction(self, other: Self, symbol: Callable[[float, float], float]) -> Self:
+        """Return the addition or subtraction of two matrices"""
+        return self.__class__(
+            tuple(tuple(symbol(n, m) for n, m in zip(row1, row2)) for row1, row2 in zip(self._matrix, other._matrix)))
+
+    def __add__(self, other: Self) -> Self:
+        """Return the addition of two matrices"""
+        return self._addition_subtraction(other, operator.add)
+
+    def __sub__(self, other: Self) -> Self:
+        """Return the subtraction of two matrices"""
+        return self._addition_subtraction(other, operator.sub)
+
+    def _matrix_multiplication(self, other: Self) -> Self:
+        """Return the multiplication of two matrices"""
+        return self.__class__(
+            tuple(tuple(sum(n * m for n, m in zip(row, col)) for col in zip(*other._matrix)) for row in self._matrix))
+
+    def __mul__(self, other) -> Self:
+        """Return the multiplication of the matrix by scalar"""
+        if isinstance(other, float):
+            return self.__class__(tuple(tuple(n * other for n in row) for row in self._matrix))

```

Comment:
move this logic outside, and do it more generic (pass the function that you want to use, for instance operator.mul)